File: javaScript_challenges/FizzBuzzTree/fizzBuzzTree.js
'use strict'

let FizzBuzzTree = tree =>{
  let arr = []
  for(let i=0; i<=tree.length-1; i++){
    if(i % 3 === 0 && i % 5 === 0){
      arr.push("FizzBuzz");
    }
    else if (i % 3 === 0){
      arr.push("Fizz");
    }
    else if (i % 5 === 0) {
      arr.push("Buzz");
    }
    else arr.push(tree[i]);;
  }
  return arr;
}

console.log(FizzBuzzTree([7,52,32,48,46,57,963,145,247,25,37]));
console.log(FizzBuzzTree([2,37,52,36,45,856,124,568,12]));
console.log(FizzBuzzTree([12,32,54,78,963,447,154,364,123]));
module.exports = FizzBuzzTree;

File: java401challenges/src/main/java/java401challenges/utilities/FizzBuzzTree.java
package java401challenges.utilities;

import java401challenges.tree.Node;
import java401challenges.tree.Tree;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.List;

public class FizzBuzzTree {


    public Node root;

    List<String> result = new LinkedList<>();

    public String traversefb(Node root) {
        if (root == null) return "The Tree is empty!";
        if (root != null) {
            if (root.value % 3 == 0 && root.value % 5 == 0) result.add("FizzBuzz");
            else if (root.value % 3 == 0) result.add("Fizz");
            else if (root.value % 5 == 0) result.add("Buzz");
            else result.add(String.valueOf(root.value));
        }
        traversefb(root.left);
        traversefb(root.right);
        return result.toString();
    }

    // Left - Root - Right
    public List<String> preOrder() {
        return this.preOrder(this.root);
    }

    public List<String> preOrder(Node node) {
        List<String> result = new LinkedList<>();
        if (node != null) {
            result.add(String.valueOf(node.value)); // Root
            result.addAll(preOrder(node.left)); // Left
            result.addAll(preOrder(node.right)); // Right
        }
        return result;
    }
}
